framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -173,6 +184,22 @@ MeshDiagnosticsGenerator::generate()
   return dynamic_pointer_cast<MeshBase>(mesh);
 }
 
+std::vector<BoundaryID>
+MeshDiagnosticsGenerator::prepareBoundaries(const std::unique_ptr<MeshBase> & mesh) const
+{
+  std::vector<BoundaryID> boundary_ids;
+  if (!_watertight_boundary_names.empty())
+  {
+    for (const auto & name : _watertight_boundary_names)
+    {
+      BoundaryID id = MooseMeshUtils::getBoundaryID(name, *mesh);
```

Comment:
you can use  MooseMeshUtils::getBoundaryIDs(...., *mesh); in the constructor maybe?